test/token/ERC20/extensions/ERC4626.test.js

```diff
@@ -44,6 +45,81 @@ contract('ERC4626', function (accounts) {
     }
   });
 
+  describe('reentrancy', async function () {
+    const reenterType = {
+      Before: 1,
+      After: 2,
+    };
+
+    let amount;
+    let token;
+    let vault;
+
+    beforeEach(async function () {
+      amount = web3.utils.toBN(10).pow(decimals);
+      token = await ERC4626ReentrantAsset.new();
+
+      vault = await ERC4626.new('', '', token.address);
+      await token.$_mint(holder, amount);
+      await token.$_mint(token.address, amount); // Requires balance to reenter
+    });
+
+    // During a `_deposit`, the vault does `transferFrom(depositor, vault, assets)` -> `_mint(receiver, shares)`
+    // such that a reentrancy BEFORE the transfer guarantees the price is kept the same.
```

Comment:
I'm not sure about this test for two reason:

- you check the price before and after the deposit that includes a re-entrancy... and there is no difference. But that is partly because the reentrant part does not change the price of the asset (a deposit doesn't affect price)
- what you want to check is that the price applied to the user is the one that was previewed... that is about it

What I would do:
- remove the `approval` part on line 74
- program the `scheduleReenter` so that it calls `token._mint(vault, some amount)` → this is an operation that actually mess up the price.
- do the preview deposit like you do
- do the deposit, and check the the event matched the preview
- do another preview deposit and observe that the price was modified by the reentrancy (but it did not affect the deposit)